.github/workflows/spike-openocd-tests.yml

```diff
@@ -14,63 +21,97 @@ jobs:
     name: Test debug (Ubuntu)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Install Dependencies
+      - name: Install packages
         run: |
-          sudo apt install build-essential device-tree-compiler
-
-#      - name: Download OpenOCD
-#        run: |
-#          git clone --recurse-submodules https://github.com/riscv/riscv-openocd.git
-#          cd riscv-openocd
-#          git checkout 43ea20dfbb6c815004a51106a3b2009d7f6c4940
+          sudo apt-get update
+          sudo apt-get install -y device-tree-compiler build-essential
 
-      - name: Build OpenOCD
+      - name: Get revisions of dependencies
         run: |
-          #cd riscv-openocd
-          ./bootstrap
-          ./configure
-          make -j"$(nproc 2> /dev/null || sysctl -n hw.ncpu)"
-          ls -l src/openocd
+          SPIKE_COMMIT=$( git ls-remote https://github.com/riscv-software-src/riscv-isa-sim.git master | awk '{ print $1; }' )
+          RISC_V_TESTS_COMMIT=$( git ls-remote https://github.com/riscv-software-src/riscv-isa-sim.git master | awk '{ print $1; }' )
+          echo "Revison of Spike: $SPIKE_COMMIT"
+          echo "Revision of RISC-V tests: $RISC_V_TESTS_COMMIT"
+          # Save for later use
+          echo "SPIKE_COMMIT=$SPIKE_COMMIT" >> $GITHUB_ENV
+          echo "RISC_V_TESTS_COMMIT=$RISC_V_TESTS_COMMIT" >> $GITHUB_ENV
 
-      - name: Download Toolchain
-        run: wget --progress=dot:giga https://github.com/xpack-dev-tools/riscv-none-elf-gcc-xpack/releases/download/v12.2.0-1/xpack-riscv-none-elf-gcc-12.2.0-1-linux-x64.tar.gz
+      - name: Get dependencies from cache
+        id: cache-deps
+        uses: actions/cache@v3
+        with:
+          path: /opt/riscv
+          key: "spike-${{env.SPIKE_COMMIT}}-toolchain-${{env.TOOLCHAIN_URL}}"
```

Comment:
Can you turn this into 2 caches: one for spike, and one for the toolchain?

The toolchain is quite large but only changes when we update this file. spike changes all the time. It seems better not to store a new copy of the toolchain just because spike changes.

Reply:
That's a good suggestion - I will do that. (Just a note: It may take me a week or so to get back to this.)